codegen/src/types/mod.rs

```diff
@@ -365,6 +365,21 @@ impl Module {
     pub fn ident(&self) -> &Ident {
         &self.name
     }
+
+    /// Returns this `Module`s child `mod`s.
+    pub fn children(&self) -> &BTreeMap<Ident, Module> {
```

Comment:
Should this (and `types` below) instead return `impl Iterator<Item = (&Ident, &Module)`? After all the `BTreeMap` usage is an implementation detail which could change.